### ISSUE TITLE
How to convert points to pixels for a non-fullscreen window ?

### ISSUE THREAD

**Issue by mulle-kybernetik-tv** [glfw/glfw#1539]
To convert font sizes from the somewhat more familiar _points_ to _pixels_, I should be using the formula `pixels = points * PPI / 72` AFAIK. But how do I get the PPI (pixel per inch/DPI) for my window ?

It would seem, I need to get the monitor first, on which my non-fullscreen window resides. And then get the monitors physical size and current resolution. The function `glfwGetWindowMonitor` will return NULL for windowed mode windows, though. 

OK, so I might have to fallback to `glfwGetPrimaryMonitor`. The physical size I could get with `glfwGetMonitorPhysicalSize`. But the video mode I get from `glfwGetVideoMode` isn't expressed as pixels but as screen coordinates, which seems possibly scaled already ? This may still be what I want, but I am not sure.

So what I have is this, converting _mm_ to _inch_:

```
monitor = glfwGetPrimaryMonitor();
glfwGetMonitorPhysicalSize( monitor, &w, &h);
mode = (GLFWvidmode *) glfwGetVideoMode( monitor);
_primaryMonitorPPI = mode->height / ((float) h * 0.03937007874);
```

Ideally of course I would just like to ask the window its PPI :)

**Comment by tombsar** [comment]
I agree it would be nice to just ask glfw for the dpi of a window. Windows' API for this actually looks sensible (assuming it works as documented): https://docs.microsoft.com/en-gb/windows/win32/api/winuser/nf-winuser-getdpiforwindow

**Comment by mulle-kybernetik-tv** [comment]
`glfwGetWindowContentScale` gives you something different. I get 1.1 for example, which is what I put in to tweak my fonts to appear larger (on Linux). PPI is not really derivable by it IMO.

**Comment by dantamont** [comment]
@mulle-kybernetik-tv Did you ever resolve how to best get the DPI you want? I have the exact same question.

**Comment by mulle-kybernetik-tv** [comment]
Nope. It's gonna bite be some time, but I am running with the naive code I outlined above for now.
